### PR TITLE
add ability to no display year, if year is the same

### PR DESCRIPTION
**Pull request by guyisra**
Sometimes it may be wanted to display a difference without a year

tell me if anything is needed more, since I quite need it (and I don't want to use my own fork)

### CODE REVIEW

**Comment by mbillard** [comment]
That seems reasonable, I'll review soon and merge if everything's all right.

File: lib/generators/time_will_tell/install/templates/config/locales/time_will_tell.en.yml [hunk 1]
```diff
@@ -2,9 +2,12 @@ en:
   time_will_tell:
     date_range:
       different_months_same_year: "%{from_month} %{from_day} %{sep} %{to_month} %{to_day}, %{year}"
+      no_year_wanted: "%{from_month} %{from_day} %{sep} %{to_month} %{to_day}"
```
Comment (mbillard) [inline lib/generators/time_will_tell/install/templates/config/locales/time_will_tell.en.yml:5]: Could you use the same format as the other keys? `different_months_no_year`

File: lib/time_will_tell/helpers/date_range_helper.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ def date_range(from_date, to_date, options = {})
         format    = options.fetch(:format, :short)
         scope     = options.fetch(:scope, 'time_will_tell.date_range')
         separator = options.fetch(:separator, '—')
+        no_year = options.fetch(:no_year, false)
```
Comment (mbillard) [inline lib/time_will_tell/helpers/date_range_helper.rb:9]: Please use `show_year` which defaults to `true` instead.

File: spec/helpers/date_range_helper_spec.rb [hunk 3]
```diff
@@ -53,6 +53,15 @@
...
     end
 
+    it 'doesnt display a year if you dont want to' do
+      from_date = Date.new(2013, 10, 31)
+      to_date   = Date.new(2013, 12, 25)
+
+      expect(
+        helper.date_range(from_date, to_date, separator: '~', no_year: true)
+      ).to eq 'Oct 31 ~ Dec 25'
+    end
```
Comment (mbillard) [inline spec/helpers/date_range_helper_spec.rb:63]: I would like a spec covering the case when the years are different and `show_year` is false where the year should still be displayed.

**Comment by mbillard** [comment]
FYI, I'm gonna think about the way the templates are used to make them more DRY because this introduces a lot of duplication.

```
different_months_same_year: "%{from_month} %{from_day} %{sep} %{to_month} %{to_day}, %{year}"
different_months_no_year: "%{from_month} %{from_day} %{sep} %{to_month} %{to_day}"
```

Instead I'd like to be able to use (not final template):

```
different_months_same_year: "%{from_month} %{from_day} %{sep} %{to_month} %{to_day}"
with_year: "%{different_months_same_year}, %{year}"
```

**Comment by mbillard** [comment]
Nevermind about the comments, I implemented the requested changes slightly differently in a different branch, it should be ready soon.

**Comment by mbillard** [comment]
Functionality is now available via the `show_year` option (set it to `false`) with version `0.1.0` of the gem. You will have to update your templates before it works.

**Comment by guyisra** [comment]
thanks a lot!

**Pull request closed without merging**